src/main/java/com/nashtech/rookie/yasa/service/user/UserServiceImpl.java

```diff
@@ -36,17 +42,24 @@ public UserDto register(RegisterDto dto) {
         user.setSecret(hashedPassword);
         user.setCart(cartService.createCart(user));
 
-        user =userRepository.save(user);
-
+        try {
+            user = userRepository.save(user);
+        }
+        catch (Exception e) {
+            throw new UserExistException();
```

Comment:
Why do you throw UserExistException here?  In this case, when it encounters an error when saving the model, it is not necessarily an error that the User already exists. It could be a connection error, an error about other invalid fields,...